### ISSUE TITLE
bug: Unable (404)  to Download the css files after create static s3 website

### ISSUE THREAD

**Issue by FelipeRuizGarcia** [localstack/localstack#5814]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I setup a s3 static website with Terraform, i am able to get the index.html, favicon etc, but not the css
at

http://myBucket.s3-website.localhost.localstack.cloud:4566/static/js/main.619cee8d.js

However, I am able to retrieve (200) the css through
http://localhost:4566/myBucket/static/js/main.619cee8d.js

Steps:

create the bucket with:
https://github.com/hashicorp/learn-terraform-modules-create/tree/main/modules/aws-s3-static-website-bucket

Upload the website to the bucket with awslocal.

I tried a lot of things with the acl, but does not work to load the css in localstack, however, those works in AWS.
```
awslocal s3 cp . s3://myBucket --recursive --include "*"  --acl public-read

```

### Expected Behavior

The S3 should return the entere website, as AWS S3 does to return all the files. 

### How are you starting LocalStack?

With the `localstack` script

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    awslocal s3 mb s3://mybucket

### Environment

```markdown
- OS: Fedora Linux 35 (Workstation Edition)
- LocalStack: 
┃ Runtime version ┃ 0.14.1                                                ┃
│ Docker image    │ tag: latest, id: 91779a2a0731, 📆 2022-03-28T10:41:10 │
│ Runtime status  │ ✔ running (name: "localstack_main", IP: 172.17.0.2)   │
```

### Anything else?

I am not sure if this is related to headers or some rule for redirects

**Comment by FelipeRuizGarcia** [comment]
Hi @MarcelStranak , did you are able to replicated the issue?

**Comment by FelipeRuizGarcia** [comment]
Updated: Still getting the issue with 
```
LocalStack CLI 0.14.2  
commit 67fddeb0e4792df58f8b61a85851b8502324015e (HEAD -> master, origin/master, origin/HEAD)
```

**Comment by MarcelStranak** [comment]
Hello, I am not sure what exactly is the issue here. I was unable to reproduce your issue.
I have taken your example and used it to deploy the static website with CSS file.
I've modified the Terraform to make it easier to deploy.
Here is the file: [example_s3_static_site.zip](https://github.com/localstack/localstack/files/8472370/example_s3_static_site.zip)
Folder structure:
```
📦aws-s3-static-website-bucket
 ┣ 📂assets
 ┃ ┗ 📜normalize.css
 ┣ 📂www
 ┃ ┣ 📜error.html
 ┃ ┗ 📜index.html
 ┣ 📜LICENSE
 ┣ 📜main.tf
 ┣ 📜outputs.tf
 ┣ 📜provider.tf
 ┣ 📜README.md
 ┗ 📜variables.tf
```
Result:
http://testbucket.s3-website.localhost.localstack.cloud:4566
![image](https://user-images.githubusercontent.com/97876429/162949953-0468f293-f869-4db9-bae5-2686fa71b509.png)

**Comment by FelipeRuizGarcia** [comment]
Hi @MarcelStranak , Thank you for the Terraform example. That really help, lets continue the debug with that zip.
I deploy the example project in a new env with localstack `0.14.2.5`.

Results:
* I am able to download the css file( get 200 and download) , I saw the config path vs module, at the terraform for assets  `s3 object` 
* The CSS are not loaded as web styles
`The stylesheet http://testbucket.s3-website.localhost.localstack.cloud:4566/assets/normalize.css was not loaded because its MIME type, “binary/octet-stream”, is not “text/css”.` 

![image](https://user-images.githubusercontent.com/14207635/165693709-1dd9782a-bf1c-4533-a447-2aebf05a3322.png)


I am still debugging to find the root cause and make it more agnostic about os system, browser, etc

**Comment by MarcelStranak** [comment]
Hi,
Yes, you are right. I did not set the content-type when uploading to S3.
You can use this list(just add the mime type if necessary):

```terraform
###WEBSITE FILES###
locals {
  content_type_map = {
    html        = "text/html",
    js          = "application/javascript",
    css         = "text/css",
    svg         = "image/svg+xml",
    jpg         = "image/jpeg",
    ico         = "image/x-icon",
    png         = "image/png",
    gif         = "image/gif",
    pdf         = "application/pdf"
  }
}
resource "aws_s3_object" "mrw-example-bucket-objects" {
  for_each = fileset("website/", "**/*.*")
  bucket   = aws_s3_bucket.mrw-example-bucket.id
  key      = each.value
  source   = "website/${each.value}"
  etag     = filemd5("website/${each.value}")
  //content_type = "text/html"
  content_type  = lookup(local.content_type_map, regex("\\.(?P<extension>[A-Za-z0-9]+)$", each.value).extension, "application/octet-stream")
}
```

**Comment by FelipeRuizGarcia** [comment]
Seems that it is fixed partially at `7a75988f`, continue debugging

**Comment by FelipeRuizGarcia** [comment]
partially fixed, struggling with svgs files at `3f2c2484`

**Comment by macnev2013** [comment]
Hello @FelipeRuizGarcia, Thank you for informing us about the issue. 

This issue has been resolved in the latest version of localstack. 
Could you possibly try again with the 'latest' version? 

Please let us know if the problem persists. Thanks.
